### ISSUE TITLE
comment-dwim Does Not Uncomment If Nesting > 1

### ISSUE THREAD

**Issue by sshaw** [nex3/haml-mode#37]
`comment-dwim` works with this:

```haml
%p Foo
```

But not with this:
```haml
%div
  %p Foo
```

Using master. 

Here's a screen capture: 

![haml-mode](https://user-images.githubusercontent.com/17570/31058448-9fb37e82-a6c1-11e7-9034-1e1e9ef2ae72.gif)

**Comment by purcell** [comment]
Yup. I _think_ it has never worked properly, partly because comments are a tricky business in `haml`: you can comment a whole nested block by adding `-#` to just the top-level node, so the idea of uncommenting the current line or region is more complex than just looking at the line beginnings.

With a bit more time this could likely be patched, but I'm really the only active committer these days, and I'm not using `haml` enough for this to be worth my time, sorry.
